test/e2e/e2e-job-template.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: template.openshift.io/v1
```

Comment:
As our goal with this project is to eventually be compatible with vanilla I would prefer not to introduce any openshift specific means if not absolutely necessary.

Reply:
Good point- also, @smarterclayton later yesterday evening pointed me to docs for the new ci-operator process which I think would provide the path forward here. I'll probably redo this PR along those lines.